web/src/features/feature-flags/api.ts

```diff
@@ -0,0 +1,40 @@
+import { useQuery } from '@tanstack/react-query';
+import { QUERY_KEYS, getBasePath, getHeaders } from 'api/helpers';
+import { FeatureFlags } from './types';
+
+export async function getFeatureFlags(): Promise<FeatureFlags> {
+  const path = '/feature-flags';
+  const requestOptions: RequestInit = {
+    method: 'GET',
+    headers: await getHeaders(path),
+  };
+
+  try {
+    const response = await fetch(`${getBasePath()}${path}`, requestOptions);
+    if (response.ok) {
+      const data = await response.json();
+      return data;
+    }
+
+    throw new Error(await response.text());
+  } catch (error) {
+    // If the request fails, we will return an empty object instead of throwing an error
+    // as the app might still be functional without feature flags
+    console.log(error);
+    return {};
+  }
+}
+
+export function useFeatureFlags(): FeatureFlags {
+  return (
+    useQuery<FeatureFlags>([QUERY_KEYS.FEATURE_FLAGS], async () => getFeatureFlags(), {
+      suspense: true,
+    }).data || {}
+  );
+}
+
+export function useFeatureFlag(name: string): boolean {
+  const features = useFeatureFlags();
+
+  return features?.[name] || false;
```

Comment:
Any reason why we don't return the key directly?
```suggestion
  return useFeatureFlags()?.[name] || false;
```

Reply:
No specific reason, I guess I personally find my version a bit more readable and easier to comprehend though :)

Reply:
I think there is a small performance advantage to returning it directly but since this is a function that won't run that often I don't think it matters much.

Go with what you prefer on both of them 🙂

Reply:
It most likely does, but even if it doesn't I think the performance cost of assigning a constant would be extremely small 😄